File: app/src/main/java/com/wavesplatform/wallet/v1/data/enums/OrderStatus.java
package com.wavesplatform.wallet.v1.data.enums;

import android.support.annotation.ColorRes;

import com.wavesplatform.wallet.R;

public enum OrderStatus {
    Accepted("Open", R.color.dex_orderbook_left_bg),
    PartiallyFilled("Partial", R.color.dex_orderbook_left_bg),
    Cancelled("Cancelled", R.color.dex_orderbook_right_bg),
    Filled("Filled", R.color.dex_orderbook_right_bg);

    private String status;
    private @ColorRes int color;

    OrderStatus(String status, @ColorRes int color) {
        this.status = status;
        this.color = color;
    }

    public String getStatus() {
        return status;
    }

    public int getColor() {
        return color;
    }
}

File: app/src/main/java/com/wavesplatform/wallet/v2/data/model/local/OrderStatus.kt
package com.wavesplatform.wallet.v2.data.model.local

import com.wavesplatform.wallet.R
import pers.victor.ext.app
import pers.victor.ext.findColor

enum class OrderStatus(val status: String, val color: Int) {
    Accepted(app.getString(R.string.my_orders_status_open), findColor(R.color.submit400)),
    PartiallyFilled(app.getString(R.string.my_orders_status_partial_filled),findColor( R.color.submit400)),
    Cancelled(app.getString(R.string.my_orders_status_canceled), findColor(R.color.error400)),
    Filled(app.getString(R.string.my_orders_status_filled), findColor(R.color.submit400))
}
